text/0071-continue-trace-over-process-boundaries.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 2023-02-01
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/71
+- RFC Status: draft
+
+# Summary
+
+To make it possible to continue a trace over process boundaries (think of one program starting another one) For this to work we will use environment variables as the carrier. An SDK should be able to fetch tracing data from environment variables on `init()` to continue a trace started in another process.
+
+# Motivation
+
+The Sentry "Cron Monitors" feature suggests that one can add something like this to the crontab:
+
+`10 * * * *   sentry-cli monitor run <monitor_id> -- python my_cron_job.py`.
+
+The `sentry-cli` call starts a trace (so it is the head of trace) and the Sentry SDK in `my_cron_job.py` should connect it's transactions to that trace.
+
+# Background
+
+## Existing trace propagation mechanism
+
+Trace propagation is: giving trace information from one service (the first service, or head of trace) to a second service, so that the second service can create transactions that are attached to the trace created in the first service.
+
+In most cases we use HTTP as the carrier for trace information. The information is propagated through two HTTP headers (`sentry-trace` and `baggage`).
+
+A few integrations (for example job queues) use queue specific meta data fields for propagating trace information. (but those implementations only propagate `sentry-trace` and maybe the legacy `tracestate` that will be rmeoved, but NOT `baggage` so some information is lost here.)
+
+The trace information can also be injected into rendered HTML as a <meta> HTML tag.
+
+See Appendix A if you want to know how this works in the Python SDK.
+
+# Options Considered
+
+## Option A)
+
+### Retrieving tracing information via environment variables:
+
+When the SDK starts up it reads an environment variable `SENTRY_TRACING_USE_ENVIRONMENT` (defaults to `False`).
```

Comment:
I think the default for `SENTRY_TRACING_USE_ENVIRONMENT` should be `True`. Otherwise most processes will not have that feature enabled and typically you control your spawning context anyways.